botocore/handlers.py

```diff
@@ -282,9 +282,11 @@ def add_expect_header(model, params, **kwargs):
         return
     if 'body' in params:
         body = params['body']
-        if hasattr(body, 'read'):
+        size = getattr(body, "_size", None)
+        if size and size != "0":
```

Comment:
```suggestion
        if size:
```

If `size` is zero, it evaluates to `false` automatically.

Reply:
That covers the case of when `size` is `"0"` as a string, so the value was not converted to `int`.